gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -1525,4 +1640,53 @@ public static AddressAggregatedListOption pageToken(String pageToken) {
    * @throws ComputeException upon failure
    */
   Operation delete(AddressId addressId, OperationOption... options);
+
+  /**
+   * Creates a new snapshot.
+   *
+   * @return a zone operation if the create request was issued correctly, {@code null} if
+   *     {@code snapshot.sourceDisk} was not found
+   * @throws ComputeException upon failure
+   */
+  Operation create(SnapshotInfo snapshot, OperationOption... options);
+
+  /**
+   * Returns the requested snapshot or {@code null} if not found.
+   *
+   * @throws ComputeException upon failure
+   */
+  Snapshot getSnapshot(String snapshot, SnapshotOption... options);
+
+  /**
+   * Lists all snapshots.
```

Comment:
+1 for removing the "all" and leaving it as "List snapshots."

@mziccard I agree with you that the description for the list should not go to detail about specific behavior based on setting but to me "all" is part of it. I think it is unnecessary in general but in particular when the service may cap the result without any settings by the user.